### PR TITLE
New UI Element Management

### PR DESCRIPTION
**Pull request by maximehuran**

### CODE REVIEW

File: src/DependencyInjection/UiElementRegistryPass.php [hunk 1]
```diff
@@ -13,28 +13,81 @@
...
+        } catch (InvalidArgumentException $exception) {
             return;
         }
 
-        $uiElementRegistry = $container->findDefinition('monsieurbiz_rich_editor.registry');
+        foreach ($uiElements as $code => $configuration) {
+            $metadataRegistry->addMethodCall('addFromCodeAndConfiguration', [$code, $configuration]);
+            $metadata = Metadata::fromCodeAndConfiguration($code, $configuration);
+
+            $id = $metadata->getServiceId('richeditor');
```
Comment (jacquesbh) [inline src/DependencyInjection/UiElementRegistryPass.php:45]: What about?
```suggestion
            $id = $metadata->getServiceId('richeditor.ui_element');
```

File: src/UiElement/Metadata.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+     * @param string $code
+     * @param array $parameters
+     */
+    private function __construct(string $code, array $parameters)
+    {
+        $this->code = $code;
+        $this->parameters = $parameters;
+    }
+
+    public static function fromCodeAndConfiguration(string $code, array $parameters): self
```
Comment (jacquesbh) [inline src/UiElement/Metadata.php:40]: Oops? phpdoc? :p

File: src/Resources/config/config.yaml [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+    monsieurbiz.text:
+      title: 'monsieurbiz_sylius_richeditor.ui_element.monsieurbiz.text.title'
+      description: 'monsieurbiz_sylius_richeditor.ui_element.monsieurbiz.text.description'
+      icon: font
+      classes:
+        form: MonsieurBiz\SyliusRichEditorPlugin\Form\Type\UiElement\TextType
+      templates:
+        admin_form: '@MonsieurBizSyliusRichEditorPlugin/UiElement/Admin/Text/adminForm.html.twig'
+        admin_render: '@MonsieurBizSyliusRichEditorPlugin/UiElement/Admin/Text/adminRender.html.twig'
+        front_render: '@MonsieurBizSyliusRichEditorPlugin/UiElement/Text/frontRender.html.twig'
```
Comment (jacquesbh) [inline src/Resources/config/config.yaml:16]: Aren't the templates missing?

**Review by jacquesbh — CHANGES_REQUESTED** [review]

File: src/Resources/config/config.yaml (hunk 3 above)
Comment (maximehuran) [reply src/Resources/config/config.yaml:16]: We will create it later

**Review by jacquesbh — APPROVED** [review]

**Pull request merged**